activestorage/lib/active_storage/attached/one.rb

```diff
@@ -54,16 +54,11 @@ def blank?
     #   person.avatar.attach(io: File.open("/path/to/face.jpg"), filename: "face.jpg", content_type: "image/jpeg")
     #   person.avatar.attach(avatar_blob) # ActiveStorage::Blob object
     def attach(attachable)
+      record.public_send("#{name}=", attachable)
       if record.persisted? && !record.changed?
-        record.public_send("#{name}=", attachable)
-        if record.save
-          record.public_send("#{name}")
-        else
-          false
-        end
-      else
-        record.public_send("#{name}=", attachable)
+        return if !record.save
```

Comment:
TBH I don't even see the reason why returning something different when the record couldn't be saved 😅 The attachment is kept in memory, so it can be saved afterward if the record is corrected..

I think `attach` should have always returned `nil` for all the cases, unfortunately, some apps might already depend on the return value..

```ruby
def attach(attachable)
  record.public_send("#{name}=", attachable)
  record.save if record.persisted? && !record.changed?
  nil
end
```

Reply:
As I mentioned in https://github.com/rails/rails/pull/45397#pullrequestreview-1011550939, I'm not sure if #44439 was motivated by a real-world use case.  It (fb88db58d7ec1c1998d90d3fdadf819200786a7c) has not been released yet, so I don't think any apps are depending on the return value.  But #44439 was merged by rafaelfranca, and #44558 was reviewed by matthewd and merged by jeremy, so I am hesitant to revert that behavior.

Reply:
There I removed those tests and I thought it'd be valuable to add a CHANGELOG entry, but let me know if you want to adjust the message or even remove it.

Apart from that, I'll work in a separate PR on adding some tests for the "attachments kept in memory" situation and find a fix for the strange scenario I found.

Let me know if I missed something.